src/app/query-explainer/query-explainer.component.ts

```diff
@@ -86,4 +100,30 @@ export class QueryExplainerComponent extends BaseComponent implements OnInit {
         );
         return commitmentUploadToken ?? "";
     }
+
+    /* istanbul ignore next */
+    public async verifyCommitment(): Promise<void> {
+        try {
+            const parsedCommitment = (await JSON.parse(this.commitment)) as QueryExplainerResponse;
+            this.commitmentUploadToken = "simulated-token";
+            this.componentData$ = combineLatest([
+                of(parsedCommitment),
+                this.queryExplainerService.processQueryWithSchema(parsedCommitment.input.query),
+                this.queryExplainerService.verifyQuery(parsedCommitment),
+            ]).pipe(
+                tap(([commitment]) => this.fillDatasetsObservables(commitment.input.datasets ?? [])),
+                map(([commitment, dataJsonAoS, sqlQueryVerify]) => {
+                    return {
+                        sqlQueryExplainerResponse: {
+                            ...commitment,
+                            ...dataJsonAoS,
+                        },
+                        sqlQueryVerify,
+                    };
+                }),
+            );
```

Comment:
The code looks duplicated

Reply:
Moved to separated method duplicated code